airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -346,7 +389,10 @@ public KubePodProcess(KubernetesClient client,
     copyFilesToKubeConfigVolume(client, podName, namespace, filesWithSuccess);
 
     LOGGER.info("Waiting until pod is ready...");
-    client.resource(podDefinition).waitUntilReady(30, TimeUnit.MINUTES);
+    client.resource(podDefinition).waitUntilCondition(p -> {
+      boolean isReady = Objects.nonNull(p) && Readiness.getInstance().isReady(p);
+      return isReady || isTerminal(p); // todo: fixes too fast completion, but what do we do for failure/error?
```

Comment:
add clarifying comment about possible error states, even though we don't think this is possible